src/main/kotlin/com/mineinabyss/deeperworld/synchronization/SectionSyncListener.kt

```diff
@@ -179,18 +177,21 @@ object SectionSyncListener : Listener {
 
     @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGH)
     fun SignChangeEvent.syncSignText() {
-        block.sync(signUpdater(lines))
+        block.sync(signUpdater(lines()))
     }
 
-    /** Disables Iron Golem and Wither summons in section transitions due to duping **/
+    /** Removes Iron Golem and Wither summons in corresponding section location due to duping **/
     @EventHandler
     fun EntitySpawnEvent.onEntitySummon() {
         if (entity.location.inSectionOverlap &&
-           (entityType == EntityType.WITHER || entityType == EntityType.IRON_GOLEM)) {
-            isCancelled = true
-            entity.location.getNearbyPlayers(5.0).forEach {
-                it.error("Spawning of $entityType is disabled in section overlaps.")
-            }
+            (entityType == EntityType.WITHER || entityType == EntityType.IRON_GOLEM)
+        ) {
+            entity.world.getNearbyEntitiesByType(
+                entityType.entityClass,
+                entity.location.correspondingLocation ?: return,
+                1.0
+            ).firstOrNull()?.remove() ?: return
```

Comment:
Should this be a remove ? Since i thought the logic was the event gets fired, checks if it was cancelled, then if it wasn't then it would spawn, this check would be before that check no?

Reply:
Previously it just cancelled the event, but made it so that the synced entity is removed and not duped